src/district/server/smart_contracts.cljs

```diff
@@ -216,12 +216,22 @@
                                                (callback nil response)
                                                (js/setTimeout #(wait-for-block block-number callback) 1000))))))
 
-(defn replay-past-events-in-order [events callback {:keys [:from-block :to-block
-                                                           :ignore-forward? :delay
-                                                           :transform-fn :on-finish]
-                                                    :or {delay 0 transform-fn identity}
-                                                    :as opts}]
-  (let [logs-chans (for [[k [contract event]] events]
+(defn replay-past-events-in-order
+  "Replay all past events in order.
+  :from-block, a block number to start replaying events from.
+  :from-tx-lidx, a touple with transaction number inside the block and log index inside the transaction.
```

Comment:
I suggest making this 3 options: 
`:from-block`
`:from-tx`
`:from-log-index`
Since, `:from-tx-lidx` is dependent on `:from-block` anyway, so can be split into 3. I think it's easier to understand. It should throw error if some options are missing.

Reply:
Yeah agree, I'll change that